sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/FeatureFlagConfigurationSetting.cs

```diff
@@ -152,51 +171,186 @@ public IList<FeatureFlagFilter> ClientFilters
         internal override void SetValue(string value)
         {
             _originalValue = value;
-
             _isValidValue = TryParseValue();
         }
 
         internal override string GetValue()
         {
-            return _isValidValue ? FormatValue() : _originalValue;
+            // If the setting was created using the composite constructor, it
+            // will not have an original value and it will need to be formatted for
+            // the first time.
+            if (_originalValue == null && _isValidValue)
+            {
+                _originalValue = CreateInitialValue();
+                return _originalValue;
+            }
+
+            // If the value wasn't valid, return it verbatim.
+            if (!_isValidValue)
+            {
+                return _originalValue;
+            }
+
+            // Form the value by coping the source JSON and replacing the setting property
+            // values.  This will ensure that any custom attributes are preserved.
+            using var memoryStream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(memoryStream);
+
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(_originalValue));
```

Comment:
Curious: could we store it in UTF8 so we don't need to go back and forth across transcodings?

Reply:
We probably could, but my intent was to make a small scope change to how we format the value.  If we want to revisit the approach used and tune efficiency, I'll ask that we open an issue and treat that as a separate effort.

Reply:
That's fair, and it was mostly an exploratory question.  I think, though, here, when we're transcoding to UTF8, manipulating, and then going back to UTF16 at the end, it might be worth considering whether there's an option to keep it in a single format the whole time.

Reply:
I'm open to ideas here, if they're in a small scope and low effort.  Otherwise, I don't believe that we have justification to take this on at present.

Reply:
To make the September release, I'm going to resolve this.   I'm happy to talk about ideas and potentially incorporate them as a follow-up.